06.Lesson_ThinkingProperties/ThinkingPropertiesExercise/MeetingScheduler/README.md

```diff
@@ -0,0 +1,79 @@
+# Rules
+
+## Takeaways
+1. proving valid states is hard when using rules because you need a bunch possibly unsound `require`s
+2. For the State Transitions when you're narrowing down the function selectors to a allowlisted set, you should use `filter` because that will be speedier
+
+## Valid States
+
+def uninitialized:
+    - startTime and endTime are 0
+    - numOfParticipants is 0
+    - organizer is 0
+    - status is UNINITIALIZED
+
+def initialized:
+    - startTime and endTime are nonzero
+    - status is not UNINITIALIZED
+
+
+
+
+1. if status is UNINITIALIZED, then state should be uninitialized
+    - probably want an invariant here
+2. if status is not UNINITIALIZED, then state should be initialized
+3. if status is ENDED, then block.timestamp must be > endTime 
+4. once status is ENDED, every non-view function should revert
+
+## State Transitions
+
+5. starting in uninitialized state, the only non-reverting non-view function should be scheduleMeeting
+6. If the state changed from UNINITIALIZED to PENDING, the only way that could happen is if `scheduleMeeting` was called
+7. If the state changed from PENDING to STARTED, the only way that could happen is if `startMeeting` was called
+8. If the state changed from STARTED to CANCELLED, the only way that could happen is if `cancelMeeting` was called
+9. If the state changed from STARTED to ENDED, the only way that could happen is if `endMeeting` was called
+
+
+## Variable Transitions
+
+10. (parametric) numParticipants should only increase or stay the same, never decrease
+
+## High-Level Properties
+
+11. (parametric) All functions that succeed for a non-organizer user would also succeed for an organizer
+12. (parametric) The status value can never decrease, only increase or stay the same
+
+## Unit Tests
+
+13. `scheduleMeeting` should only fail if state is not UNINITIALIZED, and startTime > endTime && startTime <= block.timestamp
+14. `joinMeeting` always increases numParticipants by exactly 1
+
+
+## Risk Analysis
+
+15. (parametric) any 2 function calls acting on different meetingId's should have no impact on each other
```

Comment:
From a pedagogical perspective, I think it would be ideal if we could define categories whose boundaries are less murky. I think these sorts of "it's a bit this kind of rule, but you could also call it that kind of rule" are unsatisfying and don't really give clarity to a learner about what each category is.

If we choose to present these categories, I'd really like to have some unambiguous examples for each category.